contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -14,26 +14,28 @@ contract DisputeManagerFacet is StateChannelCommon {
         bytes memory signature
     ) public { 
         // sanity checks
-        msg.sender == dispute.disputer;
+        require(msg.sender == dispute.disputer, "CREATE DISPUTE: INVALID DISPUTER");
         // race condition check
-        if(onChainSlashes.length != onChainSlashedParticipants.length 
-        || onChainSlashes[onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
-            revert("onChainSlashes does not match onChainSlashedParticipants");
+        address[] memory onChainSlashedParticipants = getOnChainSlashedParticipants();
+        if(dispute.onChainSlashes.length != onChainSlashedParticipants.length 
+        || dispute.onChainSlashes[dispute.onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
+            revert CreateDisputeInvalidOnChainSlashedParticipants();
         }
 
-        address disputer = StateChannelUtilLibrary.retriveSignerAddress(dispute.encodedBlock, signature);
+        address disputer = StateChannelUtilLibrary.retriveSignerAddress(abi.encode(dispute), signature);
         if(disputer != dispute.disputer) {
-            revert("Invalid signature");
-        }
-        if(proofs.length == 0) return;
+            revert CreateDisputeInvalidSignature();
+        } 
+        // check if fraud proofs or TimoutStruct is provided to process the dispute
+        if(dispute.fraudProofs.length == 0 || isTimeoutSetWithOptional(dispute.timeout, false)) return;
```

Comment:
You don't have to check this here, since there is no race condition on-chain for this. If the disputer commits to an invalid dispute, someone else will produce a fraud proof to slash them.